csrc/executor.cpp

```diff
@@ -1908,9 +1908,11 @@ std::vector<at::Tensor> FusionExecutor::runFusion(
   }
 
   NVF_ERROR(validKernelId(), "Invalid kernel id for FusionExecutor.");
-  NVF_ERROR(
-      !args.getCacheId().has_value() || outputs.empty(),
-      "short cut input cache is not compatible with pre-allocated output");
+
+  // short cut input cache is not compatible with pre-allocated output
+  if (!outputs.empty()) {
+    args.getCacheId() = std::nullopt;
```

Comment:
Does this mean a new `ExecutorEntry` must be created each time we launch with preallocated outputs? What if we include outputs in `args` or possibly create another `KernelArgumentHolder` that just holds the outputs, then combine the IDs?